src/tokens.rs

```diff
@@ -214,7 +212,7 @@ impl<'a> Tokenizer<'a> {
     }
 
     pub fn current(&mut self) -> usize {
-        self.chars.clone().next().map(|i| i.0).unwrap_or(self.input.len())
+        self.chars.clone().next().map(|i| i.0).unwrap_or_else(|| self.input.len())
```

Comment:
I filed Manishearth/rust-clippy#1653 for this; there's been a lot of `unwrap_or_else` false positives.